x-pack/plugins/reporting/server/routes/internal/deprecations/deprecations.ts

```diff
@@ -56,14 +51,15 @@ export const registerDeprecationsRoutes = (reporting: ReportingCore, logger: Log
     };
   };
 
+export const registerDeprecationRoutes = (reporting: ReportingCore, logger: Logger) => {
+  const { router } = reporting.getPluginSetupDeps();
+  const authzWrapper = getAuthzWrapper(reporting, logger);
+
+  const getStatusPath = INTERNAL_ROUTES.MIGRATE.GET_ILM_POLICY_STATUS;
   router.get(
-    { path: API_GET_ILM_POLICY_STATUS, validate: false },
+    { path: getStatusPath, validate: false },
```

Comment:
```suggestion
    { path: API_GET_ILM_POLICY_STATUS, validate: false, options: { access: 'internal' } },
```

I think it'd be worth using the explicit `access` option too -- technically you don't need to for internal bc we default to internal (https://github.com/elastic/kibana/pull/161672) but IMO captures intention better.

However, setting this option will be needed for paths that you intend to keep public: `options.access: 'public'` or they will be blocked on serverless (i.e. require a special header).

Reply:
Thanks JL, I have finished the other organization work that I planned for this PR, and have added the `access: public` option to the public routes here: 5122ed637b0

Reply:
@jloleysens agreed, it's really important to capture the intention and make the public and internal code and behavior clear, especially because the code looks very similar. Added the `options.access:internal` parts: 9bbc650b5aa